tests/files/tests.py

```diff
@@ -200,6 +200,23 @@ def test_shared_lock(self):
             self.assertIs(locks.unlock(f1), True)
             self.assertIs(locks.unlock(f2), True)
 
+    def test_open_supports_full_signature(self):
+        """
+        File.open supports full signature of Python's open.
+        """
+        file_path = Path(__file__).parent / "test.png"
+        mock_file = mock.Mock()
+        mock_file.name = file_path
+        mock_file.mode = "r"
+        mock_file.closed = True
+        file = File(mock_file)
+        with mock.patch(
+            "builtins.open", mock.mock_open(read_data=b"content")
+        ) as mock_open, file.open(encoding="utf-8-sig", errors="strict"):
+            mock_open.assert_called_with(
+                file_path, "r", encoding="utf-8-sig", errors="strict"
+            )
```

Comment:
Is there a reason I'm not aware of why we would want to avoid mocking?

I think mocking in a testing scenario is absolutely fine as it's intended to replace parts of the system under test with mock objects and then make assertions about how they have been used.

I think the `mock_open.assert_called_with` part in particular is very readable. A maintainer immediately knows what the intention was and what is being tested.

I would stick to the mocking approach unless there's reviewer consensus to do so differently.

Reply:
@felixxm Thoughts on this? It probably doesn't make much difference in this case but I'm (also) a fan of avoiding mocking when it's equally as easy to prove the actual integration works 👍

Reply:
I agree with David, mocking is always the last resort.